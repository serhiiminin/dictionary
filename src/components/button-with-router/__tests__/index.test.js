import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import ButtonWithRouter from '..';


describe('Button with Router', () => {
  const component = renderer.create(
    <Router>
      <ButtonWithRouter to='/anywhere'>Anywhere</ButtonWithRouter>
    </Router>
  );
  let tree = component.toJSON();

  test('render', () => {
    expect(tree).toMatchSnapshot();
  });

  test('onClick', () => {
    tree.props.onClick();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
