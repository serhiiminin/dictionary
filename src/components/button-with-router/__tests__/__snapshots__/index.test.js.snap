// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button with Router onClick 1`] = `
<button
  className="MuiButtonBase-root-27 MuiButton-root-3 ButtonCustomized-root-1 MuiButton-text-5 MuiButton-flat-8"
  disabled={false}
  location={
    Object {
      "hash": "",
      "key": "doybwe",
      "pathname": "/anywhere",
      "search": "",
      "state": undefined,
    }
  }
  match={
    Object {
      "isExact": false,
      "params": Object {},
      "path": "/",
      "url": "/",
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    Anywhere
  </span>
  <span
    className="MuiTouchRipple-root-30"
  />
</button>
`;

exports[`Button with Router render 1`] = `
<button
  className="MuiButtonBase-root-27 MuiButton-root-3 ButtonCustomized-root-1 MuiButton-text-5 MuiButton-flat-8"
  disabled={false}
  location={
    Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    }
  }
  match={
    Object {
      "isExact": true,
      "params": Object {},
      "path": "/",
      "url": "/",
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    Anywhere
  </span>
  <span
    className="MuiTouchRipple-root-30"
  />
</button>
`;
